### ISSUE TITLE
Issue in instrumenting attribute accesses in the left side of assignments.

### ISSUE THREAD

**Issue by nimamg** [sola-st/DynaPyt#62]
# Description
When instrumenting an augmented assignment, if the right side is attribute access, DynaPyt's instrumentation will lead to an incorrect code. As seen in the example below, the instrumented line results in an error (SyntaxError: 'function call' is an illegal expression for augmented assignment)

# Original Code
```python
class X:
    a = 5

x = X()

x.a += 1
```

# Instrumented Code
I am just including the instrumented `x.a += 1` line to shorten the issue.
```python
    _rt._attr_(_dynapyt_ast_, 8, x, "a") += 1
```

# Used Analysis
This can be reproduced using this simple analysis class:
```python
from dynapyt.analyses.BaseAnalysis import BaseAnalysis

class CustomAnalysis(BaseAnalysis):
    def __init__(self, output_dir=None):
        super().__init__()
        
    def read_attribute(self, dyn_ast, iid, base, name, val):
        pass
```

**Comment by nimamg** [comment]
I have also seen some instances with simple assignment (not augmented ones such as `+=`) where the attribute access bug can happen. It can also happen when using type annotations (`self.x: Optional[str] = None`) but I haven't been able to reproduce that yet, despite seeing it multiple times.

**Comment by ghminaei** [comment]
Hi,
I have the same problem. When I use my custom analysis I get a lot of "SyntaxError: 'function call' is an illegal expression for augmented assignment"s. Are there any updates on this?

**Comment by AryazE** [comment]
This should be fixed in 24f1054.
Let me know if it works.
